### PR TITLE
vector-store: fix blob primary key round-trip in CQL path

### PR DESCRIPTION
**Pull request by m-szymon**
Blob is a valid CQL primary key type. Previously, try_to_json hit unimplemented!() for CqlValue::Blob (crashing the server when a blob key appeared in an ANN result), and try_from_json had no NativeType::Blob arm (failing any ANN filter restriction on a blob-keyed column).

- try_to_json: serialise CqlValue::Blob as a '0x'-prefixed lowercase hex string, consistent with how CQL JSON represents blob values.
- try_from_json: parse '0x'-prefixed hex strings back into CqlValue::Blob so blob keys round-trip through the VS HTTP layer.
- Add unit tests covering both directions including empty blobs, single bytes, multi-byte values, and error cases (missing prefix, invalid hex, odd-length hex).
- Add blob to the serde validator test matrix, exercising the full CQL blob-key lifecycle: table creation, full-scan ingestion into InvariantKey, and ANN query results serialized via try_to_json.

Fixes: VECTOR-608

### CODE REVIEW

**Comment by m-szymon** [comment]
@QuerthDP please review.

File: crates/vector-store/src/httproutes.rs [hunk 1]
```diff
@@ -867,6 +867,16 @@ fn try_to_json(value: CqlValue) -> anyhow::Result<Value> {
             })?,
         )),
 
+        CqlValue::Blob(value) => {
+            use std::fmt::Write;
```
Comment (QuerthDP) [inline crates/vector-store/src/httproutes.rs]: I'd import it in the top of the file

File: crates/vector-store/src/httproutes.rs [hunk 2]
```diff
@@ -920,6 +930,23 @@ fn try_from_json(value: Value, cql_type: &NativeType) -> anyhow::Result<CqlValue
                 .map_err(|err| anyhow!("Failed to parse Timestamp from string '{value}': {err}"))?;
                 Ok(CqlValue::Timestamp(datetime.into()))
             }
+            NativeType::Blob => {
+                let hex = value
+                    .strip_prefix("0x")
+                    .ok_or_else(|| anyhow!("Blob value must be a '0x'-prefixed hex string"))?;
```
Comment (QuerthDP) [inline crates/vector-store/src/httproutes.rs]: Could use `.ok_or(anyhow!(...)` instead

File: crates/vector-store/src/httproutes.rs [hunk 3]
```diff
@@ -920,6 +930,23 @@ fn try_from_json(value: Value, cql_type: &NativeType) -> anyhow::Result<CqlValue
...
+                let bytes = (0..hex.len())
+                    .step_by(2)
+                    .map(|i| {
+                        u8::from_str_radix(
+                            hex.get(i..i + 2)
+                                .ok_or_else(|| anyhow!("Blob hex string has odd length"))?,
```
Comment (QuerthDP) [inline crates/vector-store/src/httproutes.rs]: ditto

File: crates/vector-store/src/httproutes.rs [hunk 4]
```diff
@@ -1639,6 +1666,42 @@ mod tests {
...
         assert!(try_to_json(CqlValue::Float(f32::NAN)).is_err());
         assert!(try_to_json(CqlValue::Double(f64::NAN)).is_err());
+
+        assert_eq!(
+            try_to_json(CqlValue::Blob(vec![0xde, 0xad, 0xbe, 0xef])).unwrap(),
+            Value::String("0xdeadbeef".to_string())
+        );
+        assert_eq!(
+            try_to_json(CqlValue::Blob(vec![])).unwrap(),
+            Value::String("0x".to_string())
```
Comment (QuerthDP) [inline crates/vector-store/src/httproutes.rs:1679]: Should it be 0x or just empty? Why?

**Review by QuerthDP — DISMISSED** [review]
Looks fine. Leaving some nitpicks and a question

**Comment by QuerthDP** [comment]
This could be opened right? Why keep it in draft?

File: crates/vector-store/src/httproutes.rs (hunk 4 above)
Comment (m-szymon) [reply crates/vector-store/src/httproutes.rs:1679]: Scylla validates the 0x prefix (throws "Blob JSON strings must start with 0x" if missing)

**Review by QuerthDP — DISMISSED** [review]

**Comment by QuerthDP** [comment]
Needs rebase for cargo deny to pass

File: crates/vector-store/src/httproutes.rs [hunk 5]
```diff
@@ -867,6 +868,15 @@ fn try_to_json(value: CqlValue) -> anyhow::Result<Value> {
             })?,
         )),
 
+        CqlValue::Blob(value) => {
+            let mut hex = String::with_capacity(2 + value.len() * 2);
+            hex.push_str("0x");
+            for b in &value {
+                write!(hex, "{b:02x}").expect("writing to String never fails");
+            }
```
Comment (ewienik) [inline crates/vector-store/src/httproutes.rs]: consider using https://docs.rs/const-hex/latest/const_hex/fn.encode_prefixed.html

File: crates/vector-store/src/httproutes.rs [hunk 6]
```diff
@@ -920,6 +930,23 @@ fn try_from_json(value: Value, cql_type: &NativeType) -> anyhow::Result<CqlValue
...
+                    .step_by(2)
+                    .map(|i| {
+                        u8::from_str_radix(
+                            hex.get(i..i + 2)
+                                .ok_or(anyhow!("Blob hex string has odd length"))?,
+                            16,
+                        )
+                        .map_err(|err| anyhow!("Invalid hex byte in blob value: {err}"))
+                    })
+                    .collect::<anyhow::Result<Vec<u8>>>()?;
```
Comment (ewienik) [inline crates/vector-store/src/httproutes.rs]: consider using https://docs.rs/const-hex/latest/const_hex/fn.decode.html

**Review by ewienik — CHANGES_REQUESTED** [review]

File: crates/vector-store/src/httproutes.rs (hunk 5 above)
Comment (m-szymon) [reply crates/vector-store/src/httproutes.rs]: ok.
scylla-cdc internally uses hex crate - it is already in dependency tree.
I went with const-hex suggestion as it reportedly has much better performance. However sending large blobs is not a recommended scenario anyway (and probably not very useful), so maybe it would be cleaner to use hex crate?

File: crates/vector-store/src/httproutes.rs (hunk 6 above)
Comment (m-szymon) [reply crates/vector-store/src/httproutes.rs]: ok.
BTW, I am not sure if we need to be so strict about accepting encoded blobs. When Scylla reads it, it requires 0x prefix. Theoretically we don't have to check it (and const_hex::decode handles it.

**Review by m-szymon — COMMENTED** [review]
Updated and rebased.

Could you check what is going on with failed (unrelated to this patch) test?

File: crates/vector-store/src/httproutes.rs (hunk 5 above)
Comment (ewienik) [reply crates/vector-store/src/httproutes.rs]: In the maintenance period we could reach different versions of the same crates, so as scylla-cdc is in different repository and project I don't think we should try to use the same dependency.

File: crates/vector-store/src/httproutes.rs (hunk 6 above)
Comment (ewienik) [reply crates/vector-store/src/httproutes.rs]: Let's be strict with checking for now.

**Review by ewienik — APPROVED** [review]
There are failed tests
